test/web/10-sessionService.js

```diff
@@ -0,0 +1,138 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+
+import delay from 'delay';
+import {SessionService} from 'bedrock-web-session';
+import {login, createAccount} from './helpers.js';
+import mockData from './mock-data.js';
+
+const sessionService = new SessionService();
+
+describe('sessionService API', () => {
+  describe('unauthenticated request', () => {
+    let session;
+    beforeEach(function() {
+      session = null;
+    });
+    afterEach(async function() {
+      await sessionService.logout();
+    });
+    it('should get a session with no account', async () => {
+      let err;
+      try {
+        session = await sessionService.get();
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(err);
+      should.exist(session);
+      session.should.be.an('object');
+      const keys = Object.keys(session);
+      // an unauthenticated session has no data
+      keys.should.deep.equal([]);
+    });
+  }); // end unauthenticated request
+  describe('authenticated request', () => {
+    let session, account, totp, email, password = null;
+    before(async function() {
+      ({account, totp, email, password} = await createAccount(
+        mockData.accounts.sessionService));
+    });
+    beforeEach(async function() {
+      session = null;
+      await login({email, password, totp});
+    });
+    afterEach(async function() {
+      await sessionService.logout();
+    });
+    it('should get a session with an account', async () => {
+      let err;
+      try {
+        session = await sessionService.get();
```

Comment:
is the code simpler if `session` is scoped to each `it`?  If one `it` is reusing `session` from a prior `it` that is a pattern to be avoided if possible.

Reply:
That's a really good idea for this test so I did that: https://github.com/digitalbazaar/bedrock-web-session/pull/9/commits/68a177b0cbcb2b52ac7eb1f27fe6acbb50cf8f26